### PR TITLE
Better image pyramid

### PR DESCRIPTION
**Pull request by fcollman**
I'd like to reopen this discussion about a better way to handle the image pyramid interface.

### CODE REVIEW

File: renderapi/image_pyramid.py [hunk 1]
```diff
@@ -37,90 +40,130 @@ def _formatUrls(self):
             d.update({'maskUrl': self.maskUrl})
         return d
 
+    def __getitem__(self, key):
```
Comment (RussTorres) [inline renderapi/image_pyramid.py]: should this also support assignment with a setitem?  The current implementation does not support 
```python
ip[0]['imageUrl'] = "file:///path/to/image"
```

File: renderapi/image_pyramid.py [hunk 2]
```diff
@@ -37,90 +40,130 @@ def _formatUrls(self):
...
+            level = int(key)
+        except ValueError as e:
+            raise RenderError("{} is not a valid mipmap level".format(key))
+        if level < 0:
+            raise RenderError(
+                "{} is not a valid mipmap level (less than 0)".format(key))
+        return "{}".format(level)
+
+    def to_dict(self):
+        return {k: v.to_dict() for k, v in self.items()}
```
[the remark this answers is not in the document]
Comment (fcollman) [reply renderapi/image_pyramid.py]: hmm... I don't think so.  Is there a reason we should?

File: renderapi/image_pyramid.py [hunk 3]
```diff
@@ -37,90 +40,130 @@ def _formatUrls(self):
...
+    maskUrl : str or None
+        uri corresponding to mask
 
-    '''
-    def __init__(self, mipMapLevels=[]):
-        self.mipMapLevels = mipMapLevels
+    """
+
+    def __init__(self, level, imageUrl=None, maskUrl=None):
+        logger.warning(
```
[the remark this answers is not in the document]
Comment (fcollman) [reply renderapi/image_pyramid.py]: yes... will change

File: renderapi/image_pyramid.py [hunk 4]
```diff
@@ -37,90 +40,130 @@ def _formatUrls(self):
...
+            raise RenderError("{} is not a valid mipmap level".format(key))
+        if level < 0:
+            raise RenderError(
+                "{} is not a valid mipmap level (less than 0)".format(key))
+        return "{}".format(level)
+
+    def to_dict(self):
+        return {k: v.to_dict() for k, v in self.items()}
+
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (fcollman) [reply renderapi/image_pyramid.py]: no reason.  did you mean to_dict or line 157?

File: renderapi/image_pyramid.py (hunk 1 above)
Comment (fcollman) [reply renderapi/image_pyramid.py]: oh i don't see why not... might actually make it more backwards compatible.

**Pull request merged**